### ISSUE TITLE
the keyboard can't input  when webview_flutter using <input type='password'> 

### ISSUE THREAD

**Issue by libo1223** [flutter/flutter#58943]
```
import 'dart:async';
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(MyApp());
}
const String kNavigationExamplePage = '''
<!DOCTYPE html><html>
<head><title>Navigation Delegate Example</title></head>
<body>
<p>
The navigation delegate is set to block navigation to the youtube website.
</p>

<input type='search'  placeholder="search"/>
<input type='password'  placeholder="password" />
<input type ='month'  />
<input type = 'file' />
</body>
</html>
''';

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
        // This makes the visual density adapt to the platform that you run
        // the app on. For desktop platforms, the controls will be smaller and
        // closer together (more dense) than on mobile platforms.
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  WebViewController _webViewController;
  String filePath = "data/user_protocol.html";

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("本地html"),
      ),
      body: Container(
        child: WebView(
          initialUrl: "",
          onWebViewCreated: ((WebViewController vc){
            _webViewController = vc;
            _loadlocalData();
          }),
        ),
      ),
    );
  }
  _loadlocalData() async {
    final String contentBase64 =
    base64Encode(const Utf8Encoder().convert(kNavigationExamplePage));
    _webViewController.loadUrl('data:text/html;base64,$contentBase64');
  }
}

```

**Comment by libo1223** [comment]
when you touch the input box which type is password, the keyboard will flash once, then will appear the other keyboard style , touch this keyboard can't input any thing.
what's worse, this error will cause the application  crash.

**Comment by libo1223** [comment]
by the way, the input has so much error. If you use my demo, you can see many functions are not available。

**Comment by libo1223** [comment]
when i touch input box,

`E/InputMethodManager: b/117267690: Failed to get fallback IMM with expected displayId=432 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{359f443 VFEDHVC.. .F...... 0,0-1080,1154}`

**Comment by VladyslavBondarenko** [comment]
Hi @libo1223 
could you provide your `flutter doctor -v`? and what webview_flutter version are you using?

about input type file, follow up on https://github.com/flutter/flutter/issues/27924

**Comment by libo1223** [comment]
``` MacdeMac-mini-2:webview_test mac$ flutter doctor -v
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale zh-Hans-CN)
    • Flutter version 1.12.13+hotfix.5 at /Users/mac/Downloads/flutter
    • Framework revision 27321ebbad (6 months ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
    • Android SDK at /Users/mac/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (1 available)
    • MIX 3 • e6e49161 • android-arm64 • Android 10 (API 29)

• No issues found! ```

**Comment by libo1223** [comment]
`webview_flutter: ^0.3.22+1`

**Comment by libo1223** [comment]
> Hi @libo1223
> could you provide your `flutter doctor -v`? and what webview_flutter version are you using?
> 
> about input type file, follow up on #27924

about input type file,I solved the problem。
now，i want to solve the problem about type='search or password'

**Comment by TahaTesser** [comment]
Hi @libo1223 
Are you using latest `webview_flutter: ^0.3.22+1`, i cannot reproduce keyboard flash using code splash on Samsung Galaxy M30

<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.19.0-4.0.pre, on Mac OS X 10.15.5 19F101, locale en-GB)
    • Flutter version 1.19.0-4.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision 2f7a59a8da (4 days ago), 2020-06-05 03:44:02 -0700
    • Engine revision d17c84e7af
    • Dart version 2.9.0 (build 2.9.0-13.0.dev 02915ec5ce)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (4 available)
    • SM M305F   • 32003c30dc19668f • android-arm64  • Android 10 (API 29)
    • macOS      • macOS            • darwin-x64     • Mac OS X 10.15.5 19F101
    • Web Server • web-server       • web-javascript • Flutter Tools
    • Chrome     • chrome           • web-javascript • Google Chrome 83.0.4103.97

• No issues found!
``` 

</details>

**Comment by Mauiie** [comment]
@libo1223 
我刚刚切换到了master 上尝试了一下，还是不行， 我是用的华为手机。 你可以换一台Android 10的手机么？我这里好几个华为和小米的Android 10都一样不能成功唤起键盘。  
现象是 键盘出现了很快有消失了。
错误日志和之前一样：

@TahaTesser 
I just switched to the master and tried it. It still didn't work. I used a Huawei phone. Can you change an Android 10 phone? I have several Huawei  and Xiaomi Android 10 phones that can't successfully evoke the keyboard.
The phenomenon is that the keyboard appears and disappears quickly.
The error log is the same as before:

webview_flutter: ^0.3.22+1

flutter doctor -v :

`[✓] Flutter (Channel master, 1.19.0-4.0.pre.128, on Mac OS X 10.15.1 19B88,
    locale zh-Hans-CN)
    • Flutter version 1.19.0-4.0.pre.128 at
      /Users/wangxin/Library/flutter/flutter
    • Framework revision 5e3e5a2a1a (9 hours ago), 2020-06-08 17:38:23 -0700
    • Engine revision 580ab9c3c5
    • Dart version 2.9.0 (build 2.9.0-13.0.dev df435c9235)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/wangxin/Library/Android/sdk
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.2.1, Build version 11B500
    • CocoaPods version 1.9.0

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[!] IntelliJ IDEA Ultimate Edition (version 2019.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[✓] Connected device (1 available)
    • ELE AL00 • GBGDU19821001370 • android-arm64 • Android 10 (API 29)`

error log ：
`E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. .F...... 0,0-1080,1971}
W/InputMethodManager(12006): startInputReason = 3
E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. .F...... 0,0-1080,1971}
I/InputMethodManager(12006): showSoftInput
E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. .F...... 0,0-1080,1971}
I/flutter (12006): ViewStateModel---constructor--->ServiceModel
E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. ......ID 0,0-1080,1971}
E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. ......ID 0,0-1080,1971}
W/InputMethodManager(12006): startInputReason = 3
I/DecorView[](12006):  old windowMode:0 new windoMode:1
W/libEGL  (12006): EGLNativeWindowType 0x6ef2e47690 disconnect failed
I/HiTouch_HiTouchSensor(12006): HiTouch restricted: Sub windows restricted.`

**Comment by Mauiie** [comment]
Thank you for your quick attention and response,  @TahaTesser 
this problem has troubled us for a long time

**Comment by libo1223** [comment]
> @libo1223
> 我刚刚切换到了master 上尝试了一下，还是不行， 我是用的华为手机。 你可以换一台Android 10的手机么？我这里好几个华为和小米的Android 10都一样不能成功唤起键盘。
> 现象是 键盘出现了很快有消失了。
> 错误日志和之前一样：
> 
> @TahaTesser
> I just switched to the master and tried it. It still didn't work. I used a Huawei phone. Can you change an Android 10 phone? I have several Huawei and Xiaomi Android 10 phones that can't successfully evoke the keyboard.
> The phenomenon is that the keyboard appears and disappears quickly.
> The error log is the same as before:
> 
> webview_flutter: ^0.3.22+1
> 
> flutter doctor -v :
> 
> ```
> [✓] Flutter (Channel master, 1.19.0-4.0.pre.128, on Mac OS X 10.15.1 19B88,
>     locale zh-Hans-CN)
>     • Flutter version 1.19.0-4.0.pre.128 at
>       /Users/wangxin/Library/flutter/flutter
>     • Framework revision 5e3e5a2a1a (9 hours ago), 2020-06-08 17:38:23 -0700
>     • Engine revision 580ab9c3c5
>     • Dart version 2.9.0 (build 2.9.0-13.0.dev df435c9235)
>     • Pub download mirror https://pub.flutter-io.cn
>     • Flutter download mirror https://storage.flutter-io.cn
> 
>  
> [!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
>     • Android SDK at /Users/wangxin/Library/Android/sdk
>     • Platform android-29, build-tools 28.0.3
>     • Java binary at: /Applications/Android
>       Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build
>       1.8.0_212-release-1586-b4-5784211)
>     ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
>       --android-licenses
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 11.2.1, Build version 11B500
>     • CocoaPods version 1.9.0
> 
> [✓] Android Studio (version 3.6)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin version 45.1.1
>     • Dart plugin version 192.8052
>     • Java version OpenJDK Runtime Environment (build
>       1.8.0_212-release-1586-b4-5784211)
> 
> [!] IntelliJ IDEA Ultimate Edition (version 2019.1.2)
>     • IntelliJ at /Applications/IntelliJ IDEA.app
>     ✗ Flutter plugin not installed; this adds Flutter specific functionality.
>     ✗ Dart plugin not installed; this adds Dart specific functionality.
>     • For information about installing plugins, see
>       https://flutter.dev/intellij-setup/#installing-the-plugins
> 
> [✓] Connected device (1 available)
>     • ELE AL00 • GBGDU19821001370 • android-arm64 • Android 10 (API 29)`
> ```
> 
> error log ：
> 
> ```
> E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. .F...... 0,0-1080,1971}
> W/InputMethodManager(12006): startInputReason = 3
> E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. .F...... 0,0-1080,1971}
> I/InputMethodManager(12006): showSoftInput
> E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. .F...... 0,0-1080,1971}
> I/flutter (12006): ViewStateModel---constructor--->ServiceModel
> E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. ......ID 0,0-1080,1971}
> E/InputMethodManager(12006): b/117267690: Failed to get fallback IMM with expected displayId=463 actual IMM#displayId=0 view=io.flutter.plugins.webviewflutter.InputAwareWebView{3691351 VFED..C.. ......ID 0,0-1080,1971}
> W/InputMethodManager(12006): startInputReason = 3
> I/DecorView[](12006):  old windowMode:0 new windoMode:1
> W/libEGL  (12006): EGLNativeWindowType 0x6ef2e47690 disconnect failed
> I/HiTouch_HiTouchSensor(12006): HiTouch restricted: Sub windows restricted.
> ```

我上面贴了flutter doctor -v，我用的是小米mix3 android10版本。

**Comment by libo1223** [comment]
hi @TahaTesser 
yes, i using latest webview_flutter: ^0.3.22+1. by the way , when i use virtual machine, the keyboard cannot flash, also ,it can input. 
and can you input anything and appear * on the input box?

**Comment by libo1223** [comment]
I tested some phones, meizu  PRO 7 Plus  android 7,  huawei and onePlus.

1. meizu PRO 7Plus android 7. when touch the password, the ui will flash once .it can input.

2. huawei, touch the password won't flash, but can't input. if you touch search first and then touch password, it can input.

3. onePlus 6  android10 , touch the password will flash once , can't input  even though touch search first like my phone which mix3 android10.

**Comment by libo1223** [comment]
@TahaTesser @VladyslavBondarenko i think i found the reason。i tested Samsung, it works very well. I think that's why some systems overwrite keyboards. So can you give a api for us to choose
keyboards?

**Comment by libo1223** [comment]
@Mauiie 我下午又测了一些机型，感觉是因为国产厂商魔改了键盘，但是flutter在唤起的时候唤不起正常的键盘，感觉是这里起冲突了。然后安卓版本也很重要。安卓10之前的版本都是可以通过先点击一个input，然后点击password的input可以解决输入问题，安卓10我还没找到方法可以输入。
然后我选用了三星的手机发现，发现password和search出现的键盘都是正常的。也是可以进行正常的操作。

**Comment by Mauiie** [comment]
@libo1223  你有没有发现 webview_flutter 里面上线滚动页面的时候，Android上会感觉有些卡顿，看着眼睛很不舒服？   还是你们已经解决了?   方便的话咱加个好友吧  1203407263  qq

**Comment by libo1223** [comment]
> @libo1223 还有一个问题，不知道你发现没，webview_flutter 里面加载的网页，在滑动的时候感觉有点卡、不流畅、长时间看眼睛不舒服，特别是文字能明显感觉出来。
> 今天我还试着用X5的内核改造了webview_flutter,整体感觉也一般，他在Android10上密码输入框点击直接闪退，报错还是一样 .... 另外我加了你QQ，翻遍的话通过下我们先交流交流

980160988，我这边没有收到，你在添加一下试试

**Comment by darshankawar** [comment]
Hi @libo1223 I am unable to replicate this issue on `dev channel (1.20.0-0.0.pre)`. 

When I tap on `password` field, the keyboard opens up properly and I can type in there.

<details>
<summary>flutter doctor -v</summary>
```
[✓] Flutter (Channel dev, 1.20.0-0.0.pre, on Mac OS X 10.15.2 19C57, locale en-US)
    • Flutter version 1.20.0-0.0.pre at /Users/deeptibelsare/Documents/Fluttersdk/flutter
    • Framework revision d9653445f4 (8 days ago), 2020-06-09 18:43:03 -0400
    • Engine revision e8c13aa012
    • Dart version 2.9.0 (build 2.9.0-14.0.dev 5c1376615e)

 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/deeptibelsare/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/deeptibelsare/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/deeptibelsare/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions.

 
[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.9.3

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)

! Doctor found issues in 1 category.

```
</details>

**Comment by libo1223** [comment]
Sorry, it can't solve my problem. if you have xiaomi or huawei, you will find the issue
@darshankawar

**Comment by libo1223** [comment]
使用SingleChildScrollView包裹webview可以解决小米和华为的安全键盘问题。

**Comment by benqy** [comment]
> 使用SingleChildScrollView包裹webview可以解决小米和华为的安全键盘问题。

太感谢了, 兄弟你解决了困扰了我几个世纪的难题.

**Comment by allan-hx** [comment]
请问这个解决了么，我的添加SingleChildScrollView直接报错了，我尝试使用PlatformView自己封装webview也有类似问题

**Comment by benqy** [comment]
> 请问这个解决了么，我的添加SingleChildScrollView直接报错了，我尝试使用PlatformView自己封装webview也有类似问题

其它可以,华为下还是不行,一弹出安全键盘,键盘以上区域就白屏. 报错倒是没有

**Comment by allan-hx** [comment]
@benqy 你是使用了SingleChildScrollView包裹么？能发个代码片段我看下你的SingleChildScrollView是怎么包裹的么？我使用SingleChildScrollView包裹报错

**Comment by benqy** [comment]
> @benqy 你是使用了SingleChildScrollView包裹么？能发个代码片段我看下你的SingleChildScrollView是怎么包裹的么？我使用SingleChildScrollView 的child指定高度, flutter我用得很不熟, 各组件都用得不规范. InsideView是自己封装过的webview_flutter
`
 Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
        appBar: AppBar(
          backgroundColor: Constants.mainColor,
          title: Text('登录'),
        ),
        body: SingleChildScrollView(
          child: Container(
            height: MediaQuery.of(context).size.height,
            child: InsideView(
              url: 'https://xxx.xxx.xxx/xxx',
              onPageFinished: (url, controller) async {
                await controller.evaluateJavascript("document.querySelector('.qq-login').remove()");
                checkLogin(context, controller);
                return Future<String>.value('done');
              },
            ),
          ),
        ));
  }
`

**Comment by allan-hx** [comment]
> > @benqy 你是使用了SingleChildScrollView包裹么？能发个代码片段我看下你的SingleChildScrollView是怎么包裹的么？我使用SingleChildScrollView 的child指定高度, flutter我用得很不熟, 各组件都用得不规范. InsideView是自己封装过的webview_flutter
> 
> ```
>  Widget build(BuildContext context) {
>     return Scaffold(
>       backgroundColor: Colors.white,
>         appBar: AppBar(
>           backgroundColor: Constants.mainColor,
>           title: Text('登录'),
>         ),
>         body: SingleChildScrollView(
>           child: Container(
>             height: MediaQuery.of(context).size.height,
>             child: InsideView(
>               url: 'https://xxx.xxx.xxx/xxx',
>               onPageFinished: (url, controller) async {
>                 await controller.evaluateJavascript("document.querySelector('.qq-login').remove()");
>                 checkLogin(context, controller);
>                 return Future<String>.value('done');
>               },
>             ),
>           ),
>         ));
>   }
> ```

谢谢，应该是我高度设置问题

**Comment by libo1223** [comment]
> > 请问这个解决了么，我的添加SingleChildScrollView直接报错了，我尝试使用PlatformView自己封装webview也有类似问题
> 
> 其它可以,华为下还是不行,一弹出安全键盘,键盘以上区域就白屏. 报错倒是没有

华为白屏这个没有办法的，我估计是滚动布局在键盘弹出的时候，布局重新加载了一下，导致加载失败。而且现在这个解决方案和百度输入法起冲突，我已经将项目里的webview都改为原生webview展示了，这坑没法填了。

**Comment by blasten** [comment]
I don't see the issue with input[type=password], but I do see the issue with [type=month] and [type=file]

**Comment by LailaiMaster** [comment]
@libo1223 华为误我啊 难受

**Comment by yaphone** [comment]
Got the same problem :(

**Comment by LailaiMaster** [comment]
I tried " webview_flutter”and“ flutter_inappwebview"， both have problems.
HUAWEI phone ‘Secure input’ is open,when h5 <input > type is"password",the Secure keyboard will be opened，and webview will be hide，how to fix？

**Comment by allan-hx** [comment]
> I don't see the issue with input[type=password], but I do see the issue with [type=month] and [type=file]



> I don't see the issue with input[type=password], but I do see the issue with [type=month] and [type=file]

This problem will not appear on the simulator. It should only appear in those systems with modified input methods in China. You can try to install the third-party input method and then test it

**Comment by LailaiMaster** [comment]
#61133 fix

**Comment by t-walker-wei** [comment]
> Sorry, it can't solve my problem. if you have xiaomi or huawei, you will find the issue
> @darshankawar



> > I don't see the issue with input[type=password], but I do see the issue with [type=month] and [type=file]
> 
> > I don't see the issue with input[type=password], but I do see the issue with [type=month] and [type=file]
> 
> This problem will not appear on the simulator. It should only appear in those systems with modified input methods in China. You can try to install the third-party input method and then test it

I tested on Xiaomi9, the security keyboard flash and then the system show the normal keyboard for password input fields.But when I tested on Blackberry Key one, no security keyboard flash, just show the normal keyboard, cannot input either.

**Comment by blasten** [comment]
This has been fixed in Flutter 1.22/ WebView 1.0 - use hybrid composition in the Flutter WebView plugin.

https://github.com/flutter/plugins/tree/master/packages/webview_flutter#using-hybrid-composition

**Comment by yexiaodong** [comment]
我是用Flutter 1.22.2和webview_flutter: ^1.0.5，在小米手机上能正常显示键盘，但是会有闪烁（先黑屏一下）。另外input失去焦点后，键盘没有消失

**Comment by libo1223** [comment]
> 我是用Flutter 1.22.2和webview_flutter: ^1.0.5，在小米手机上能正常显示键盘，但是会有闪烁（先黑屏一下）。另外input失去焦点后，键盘没有消失

现在这个好像已经修复了，可以升级下webview版本再试一下。或者可以参考这个大佬的改法，https://github.com/yenole/plugins/commit/3d7b70be2a9c30ffd99dc13ba2e772ce423a8402

**Comment by yexiaodong** [comment]
1.0.5已经是最新的了

**Comment by libo1223** [comment]
> 1.0.5已经是最新的了

那你试下这个大佬的改法，我们很早就该用原生webview了，很久没关注了。这个大佬的改法在当初解决了我们的问题。

**Comment by andyxiaoxi** [comment]
> > 请问这个解决了么，我的添加SingleChildScrollView直接报错了，我尝试使用PlatformView自己封装webview也有类似问题
> 
> 其它可以,华为下还是不行,一弹出安全键盘,键盘以上区域就白屏. 报错倒是没有

我也遇到同样的问题，请问你怎么解决的？

**Comment by andyxiaoxi** [comment]
> webview

大佬，为什么，我用Androidview 实现的安桌原生webview，也会有白屏问题

**Comment by libo1223** [comment]
> > webview
> 
> 大佬，为什么，我用Androidview 实现的安桌原生webview，也会有白屏问题

qq 980160988

**Comment by libo1223** [comment]
> > > 请问这个解决了么，我的添加SingleChildScrollView直接报错了，我尝试使用PlatformView自己封装webview也有类似问题
> > 
> > 
> > 其它可以,华为下还是不行,一弹出安全键盘,键盘以上区域就白屏. 报错倒是没有
> 
> 我也遇到同样的问题，请问你怎么解决的？

https://github.com/flutter/flutter/issues/58943#issuecomment-668404917

**Comment by andyxiaoxi** [comment]
> > > webview
> > 
> > 
> > 大佬，为什么，我用Androidview 实现的安桌原生webview，也会有白屏问题
> 
> qq 980160988

我加你了

**Comment by EdoardoVignati** [comment]
Hi, the problem happens also on my Huawei P smart using flutter_inappwebview also with <input type='text'>
I found that the problem is related only to the Microsoft Swiftkey keyboard
Here the logs of the error:

```
D/OpenGLRenderer( 3382):   HWUI Binary is  enabled
W/InputMethodManager( 3382): startInputReason = 3
W/IInputConnectionWrapper( 3382): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper( 3382): endBatchEdit on inactive InputConnection
W/IInputConnectionWrapper( 3382): getExtractedText on inactive InputConnection
W/IInputConnectionWrapper( 3382): getTextBeforeCursor on inactive InputConnection
I/HiTouch_HiTouchSensor( 3382): HiTouch restricted: Sub windows restricted.
D/HiTouch_PressGestureDetector( 3382): onAttached, package=it.xyz, windowType=2037, mHiTouchRestricted=true
D/mali_winsys( 3382): EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
```

In particular take a look to: _HiTouch restricted: Sub windows restricted._

This is what I found. Any idea or workaround?

**Comment by TahaTesser** [comment]
Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by Jvaeyhcd** [comment]
> This has been fixed in Flutter 1.22/ WebView 1.0 - use hybrid composition in the Flutter WebView plugin.
> 
> https://github.com/flutter/plugins/tree/master/packages/webview_flutter#using-hybrid-composition

Thanks, It's solved my problem!!
